### PR TITLE
chore: bump checkers typos version so it works on mac and fix typos

### PR DESCRIPTION
**Pull request by dpcollins-google**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7693)

### CODE REVIEW

File: .typos.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "google/cloud/storage/client_options_test.cc",
+    "google/cloud/storage/client_sign_policy_document_test.cc",
+    "google/cloud/storage/client_sign_url_test.cc",
+    "google/cloud/storage/oauth2/google_credentials_test.cc",
+    "google/cloud/storage/oauth2/service_account_credentials_test.cc",
+    "google/cloud/storage/testing/constants.h",
+    "google/cloud/storage/tests/test_service_account.not-a-test.json",
+    "google/cloud/storage/tests/unified_credentials_integration_test.cc",
+    "google/cloud/storage/tools/make_jwt_assertion_for_test_data.py",
+]
```
Comment (coryan) [inline .typos.toml:16]: nit: newline at end of file.

File: google/cloud/internal/base64_transforms_test.cc [hunk 2]
```diff
@@ -68,12 +68,12 @@ TEST(Base64, RFC4648TestVectors) {
   // https://tools.ietf.org/html/rfc4648#section-10
   std::vector<std::pair<std::string, std::string>> test_cases = {
       {"", ""},
-      {"f", "Zg=="},
-      {"fo", "Zm8="},
-      {"foo", "Zm9v"},
-      {"foob", "Zm9vYg=="},
-      {"fooba", "Zm9vYmE="},
-      {"foobar", "Zm9vYmFy"},
+      {"a", "Zg=="},
```
Comment (coryan) [inline google/cloud/internal/base64_transforms_test.cc]: you can use `/bin/echo -n "a" | openssl base64 -e` to get the matching strings.

**Review by devjgm — APPROVED** [review]
Thanks!

**Review by coryan — DISMISSED** [review]

File: google/cloud/internal/base64_transforms_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/internal/base64_transforms_test.cc]: Oh, I think you fixed that already

**Review by coryan — APPROVED** [review]
Thanks!

**Pull request merged**